src/float_parser.ts

```diff
@@ -0,0 +1,39 @@
+import BN from "bn.js";
+
+const buf = new ArrayBuffer(8);
+const view = new DataView(buf);
+
+export function numberToFraction(number: number) {
+  // retrieve mantissa, exponent, and sign from bit representation of number
+  view.setFloat64(0, number);
+  const bits = view.getBigUint64(0);
+  const sign = (bits >> BigInt(63)) ? BigInt(-1) : BigInt(1);
```

Comment:
Does TS not work the the `n` notation (like `63n`)? It might just be a case of a missing flag in the `tsconfig` file.

Reply:
Why do we need BigInts here rather than regular ints?

Reply:
From my limited knowledge of JS, it only uses 32-bit integers for doing bitwise operations, otherwise all numbers are 64-bit floats. This means that there is no way to correctly represent 64-bit integer which is what is needed here for the bitwise operations.

Reply:
I could set `"target": "esnext"` instead of the current `"target": "es5"` in `tsconfig.json` to get the `n` notation working. But I'd say this should be done in case in another PR.
Just as a confirmation to what Nick said: `2**31 << 1 === 0`.